includes/class-wc-stripe-apple-pay.php

```diff
@@ -486,6 +486,9 @@ public function generate_apple_pay_single() {
 			if ( 'simple' === ( version_compare( WC_VERSION, '3.0.0', '<' ) ? $product->product_type : $product->get_type() ) ) {
```

Comment:
`$product->is_type('simple')` could replace this whole line and be backcompatible